astropy/utils/tests/test_misc.py

```diff
@@ -60,11 +69,13 @@ def test_skip_hidden():
         # break after the first level since the data dir contains some other
         # subdirectories that don't have these files
         break
-
-    for root, dirs, files in misc.walk_skip_hidden(path):
-        assert ".hidden_file.txt" not in files
-        assert "local.dat" in files
-        break
+    with pytest.warns(
+        AstropyDeprecationWarning, match="^The walk_skip_hidden function is deprecated"
+    ):
+        for root, dirs, files in misc.walk_skip_hidden(path):
```

Comment:
Hmmm! `extension-helpers` is only needed for building the package, not installing the package. So, what happens to this test when we remove the deprecated function?

Reply:
This test isn't useful anymore, so it should be removed together with the deprecated functions.